src/parser.rs

```diff
@@ -297,6 +300,97 @@ fn parse_attrs(
     })
 }
 
+fn next_token_is_doc(token: &TokenTree) -> bool {
+    match token {
+        TokenTree::Group(ref g) => g.to_string().starts_with("[doc = "),
+        _ => false,
+    }
+}
+
+fn stringify_tokens(tokens: TokenStream) -> String {
+    lazy_static! {
+        static ref BLOCK_OUTER_DOC_RE: Regex =
+            Regex::new(r#"^\[doc = "(?:/\*\*)(.*)(?:\*/)"\]$"#).unwrap();
+        static ref BLOCK_INNER_DOC_RE: Regex =
+            Regex::new(r#"^\[doc = "(?:/\*!)(.*)(?:\*/)"\]$"#).unwrap();
+        static ref LINE_DOC_RE: Regex = Regex::new(r#"^\[doc = "(.*)"\]$"#).unwrap();
+    }
+    let mut s = String::new();
+    let mut iter = tokens.into_iter().peekable();
+    while let Some(tok) = iter.next() {
+        match tok {
+            TokenTree::Punct(ref punct) => {
+                if punct.as_char() == '!' && iter.peek().map(next_token_is_doc).unwrap_or(false) {
+                    // remove last `# ` by calling pop twice
+                    s.pop();
+                    s.pop();
```

Comment:
Could you add assertions that poped values are correct?

Reply:
Thank you for pointing out.
I have fixed it.
During my fixing, I noticed that it's enough to call `pop` just once. 

Change lists are:

- Call `pop` just once, not twice.
- Add an assertion to check if the popped character is `#`.
- Add a comment explaining why calling `pop` is required.
- Change a variable name `s` -> `res` (tiny refactoring)